src/ibek/params.py

```diff
@@ -33,74 +33,73 @@ def __repr__(self):
         return str(self.value)
 
 
-class Value(BaseSettings):
+class Define(BaseSettings):
```

Comment:
OK, I now realise I don't like this name. Classes shouldn't be verbs. And `DefinesTypes` is very unclear.

It also leads me to a question. Are `pre_defines` and `post_defines` different to `Definitions`?

Reply:
yes this was one of the points in slack we have definitions that contain pre_defines and post_defines.

Your verb point is also good.

So how about values then!! ??

Reply:
i.e.

pre_values
post_values

Reply:
Should they not be called pre/post args given that they are inserted into the Entity args?

Reply:
I'm not that keen on the trend we have started of using different class names in the code than the key names in yaml.
Maybe we could fix that and this issue by renaming
defs: 
entityDefinitions:

which are what we called the class in the code.

Reply:
So no we have `entityDefinitions` in support yaml defining what `entities` look like in ioc.yaml.
These are `EntityDefinition` and `Entity` classes respectively in the code.

Reply:
Then maybe its OK to keep `pre_defines` and `post_defines`.

These are distinct from Parameters (no longer called Args and now appearing under `params"` in yaml) because they are calculated - never supplied by writer of an ioc.yaml - thus distinctly not parameters (or args!)

Reply:
EntityDefinition in yaml and code sounds good.

> These are distinct from Parameters (no longer called Args and now appearing under params: in yaml) because they are calculated - never supplied by writer of an ioc.yaml - thus distinctly not parameters (or args!)

Ah right they are being renamed here. Then should [this](https://github.com/epics-containers/ibek/pull/230/files#diff-2f5f49203dc0a625d87caa35446dc0c6ea88a2338858b708d62c0874ed2bb86cR70) be updated? Or are there now Args and Params?

It looks like (what is currently called) Defines are converted directly into the EntityFactory args (just a dictionary of builtins) and then that is used to populate the fields of the dynamically generated Entity model, which I think makes then parameters, right?

Reply:
Yes all use of 'arg' in the code should be replaced with param and your link is to an oversight.

You are right that 'defines' get treated like 'params' because they go in the Entity Model. But you are not supposed to be able to use them as params - now I'm worried that you can - just checking that now ...

Reply:
Yes - you can reference definitions in the ioc.yaml so that was unintentional.

Now the question is is that a bad thing. i.e. although they are intended as calculated values or constants, you can still override them in the ioc.yaml if you wish.

I think that probably it is a bad thing because the order of interpretation of jinja strings is supposed to be clear and I'm feeling that this is muddying it. Also - I did not want to confuse the ioc instance writer with many extra parameters that they should not normally set.

So I need to fix that and then still come up with the correct name.

I used pre_defines and post_defines because they are SUPPOSED to be like defines in C ie. calculated values or constants.

Reply:
@GDYendell please can we spend 20 mins IRL discussing this today as I'd like to resolve ASAP.